### PR TITLE
[ENH] Add AEBiGRUNetwork

### PR DESCRIPTION
**Pull request by aadya940**
Implements Auto-Encoder based on BiDirectional GRU Network.

### CODE REVIEW

File: aeon/networks/_ae_bgru.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+from aeon.networks.base import BaseDeepNetwork
+
+
+class AEBiGRUNetwork(BaseDeepNetwork):
+    """
+    A class to implement an Auto-Encoder based on Bidirectional GRUs.
+
+    Parameters
+    ----------
+        latent_space_dim : int, default=128
```
Comment (hadifawaz1999) [inline aeon/networks/_ae_bgru.py:12]: temporal_latent_space, default False, and if set to True, make sure that the output of the encoder is temporal series

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]

File: aeon/networks/tests/test_ae_bgru.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""Tests for the AEBiGRU Model."""
+
+import random
+
+import pytest
+
+from aeon.networks import AEBiGRUNetwork
```
Comment (hadifawaz1999) [inline aeon/networks/tests/test_ae_bgru.py:7]: would be good to test if when temporal_latent_space is True does the space really become temporal or not, just checking the shapes i mean

Comment (hadifawaz1999) [reply aeon/networks/tests/test_ae_bgru.py:7]: can leave that for later though :)
wanna get the networks in

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
this is good to go, just need the test to assert the shapes of input and output of the network are the same

**Review by hadifawaz1999 — APPROVED** [review]
lgtm

**Comment by aadya940** [comment]
> this is good to go, just need the test to assert the shapes of input and output of the network are the same

Already done in `test_all_networks.py`.

**Comment by aadya940** [comment]
There's a mistake here.
`_config.structure` should be "auto-encoder" not "encoder".

**Comment by aadya940** [comment]
#1962

**Pull request merged**